src/NuGet.Core/NuGet.Build.Tasks/NuGet.targets

```diff
@@ -89,7 +89,7 @@ Copyright (c) .NET Foundation. All rights reserved.
     Main entry point for restoring packages
     ============================================================
   -->
-  <Target Name="Restore" DependsOnTargets="_GenerateRestoreGraph">
+  <Target Name="Restore" DependsOnTargets="_GenerateRestoreGraph" Returns="$(RestoreChangedImportsProjectList)">
```

Comment:
Consider adding a comment to the targets about this, it is covered in the task but users don't always see that source.